packages/common/http/src/headers.ts

```diff
@@ -264,16 +267,16 @@ export class HttpHeaders {
 
 /**
  * Verifies that the headers object has the right shape: the values
- * must be either strings or arrays. Throws an error if an invalid
+ * must be either strings, numbers or arrays. Throws an error if an invalid
  * header value is present.
  */
 function assertValidHeaders(headers: Record<string, unknown>):
-    asserts headers is Record<string, string|string[]> {
+    asserts headers is Record<string, string|string[]|number|number[]> {
   for (const [key, value] of Object.entries(headers)) {
-    if (typeof value !== 'string' && !Array.isArray(value)) {
+    if (!['string', 'number'].includes(typeof value) && !Array.isArray(value)) {
```

Comment:
```suggestion
    if ((typeof value === 'string' || typeof value === 'number') && !Array.isArray(value)) {
```
Direct comparisons of `typeof x === 'y'` are highly optimized in the VM, whereas creating an array just to check `.includes` is quite expensive in comparison.

Reply:
fixed!